demos/mqtt/mqtt_demo_basic_tls/mqtt_demo_basic_tls.c

```diff
@@ -370,367 +357,95 @@ static void cleanupOutgoingPublishWithPacketID( uint16_t packetId );
  * the broker. This function handles the resending of the QoS2 publish packets,
  * which are maintained locally.
  *
- * @param[in] pContext MQTT context pointer.
+ * @param[in] pMqttContext MQTT context pointer.
  */
-static int handlePublishResend( MQTTContext_t * pContext );
+static int handlePublishResend( MQTTContext_t * pMqttContext );
 
 /*-----------------------------------------------------------*/
 
-static int connectToServer( const char * pServer,
-                            uint16_t port,
-                            int * pTcpSocket )
+static int connectToServerWithBackoffRetries( NetworkContext_t * pNetworkContext )
 {
-    int status = EXIT_SUCCESS;
-    struct addrinfo hints, * pIndex, * pListHead = NULL;
-    struct sockaddr * pServerInfo;
-    uint16_t netPort = htons( port );
-    socklen_t serverInfoLength;
-    struct timeval transportTimeout;
-
-    /* Add hints to retrieve only TCP sockets in getaddrinfo. */
-    ( void ) memset( &hints, 0x00, sizeof( hints ) );
-    /* Address family of either IPv4 or IPv6. */
-    hints.ai_family = AF_UNSPEC;
-    /* TCP Socket. */
-    hints.ai_socktype = SOCK_STREAM;
-    hints.ai_protocol = IPPROTO_TCP;
-
-    /* Perform a DNS lookup on the given host name. */
-    status = getaddrinfo( pServer, NULL, &hints, &pListHead );
-
-    if( status != -1 )
+    int returnStatus = EXIT_SUCCESS;
+    bool backoffSuccess = true;
```

Comment:
Would a better name be `retryAttemptsRemaining`?

Reply:
I do agree that a better name is needed but I feel that `retryAttemptsRemaining` sounds like a quantity rather than a boolean. I am going to go with `retriesArePending` instead, but let me know if you disagree.